nomos-services/data-availability/sampling/src/lib.rs

```diff
@@ -697,6 +779,33 @@ where
                             sampler.handle_sampling_error(blob_id).await;
                         }
                     }
+
+                    Some(blob_id) = blob_stream.next() => {
+                        Self::handle_incoming_blob(
+                            blob_id,
+                            sdp_blob_trigger_sampling_delay,
+                            &PendingTasks {
+                                long_tasks: &mut long_tasks,
+                                sampling_continuations: &mut sampling_continuations,
+                                delayed_sdp_sampling_triggers: &mut delayed_sdp_sampling_triggers,
+                            }
```

Comment:
Wouldn't it be better to pack it before hand and just share the &mut to the created `PendingTasks` instead of creating an object each time?